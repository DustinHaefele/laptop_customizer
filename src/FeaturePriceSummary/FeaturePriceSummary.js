import React from "react";
import FeaturePrice from "../FeaturePrice/FeaturePrice";

function FeaturePriceSummary(props) {
  const total = Object.keys(props.selected).reduce(
    (acc, curr) => acc + props.selected[curr].cost,
    0
  );

  return (
    <section className="main__summary">
      <h3>NEW GREENLEAF 2018</h3>

      {Object.keys(props.selected).map(key => {
        return <FeaturePrice key={key} 
          cost={props.selected[key].cost} 
          name={props.selected[key].name} />;
      })}

      <div className="summary__total">
        <div className="summary__total__label">Your Price: </div>
        <div className="summary__total__value">
          {new Intl.NumberFormat("en-US", {
            style: "currency",
            currency: "USD"
          }).format(total)}
        </div>
      </div>
    </section>
  );
}

export default FeaturePriceSummary;
