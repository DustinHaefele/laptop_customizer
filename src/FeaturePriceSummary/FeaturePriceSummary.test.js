import React from "react";
import ReactDOM from "react-dom";
import Enzyme from "enzyme";
import toJson from "enzyme-to-json";
import FeaturePriceSummary from "./FeaturePriceSummary";

describe("Price Summary Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<FeaturePriceSummary 
      selected={{
        Processor: {
          name: "17th Generation Intel Core HB (7 Core with donut spare)",
          cost: 700
        },
        "Operating System": {
          name: "Ubuntu Linux 16.04",
          cost: 200
        },
        "Video Card": {
          name: "Toyota Corolla 1.5v",
          cost: 1150.98
        },
        Display: {
          name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
          cost: 1500
        }
      }} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected',() => {
    const wrapper = Enzyme.shallow(
      
      <FeaturePriceSummary 
      selected={{
        Processor: {
          name: "17th Generation Intel Core HB (7 Core with donut spare)",
          cost: 700
        },
        "Operating System": {
          name: "Ubuntu Linux 16.04",
          cost: 200
        },
        "Video Card": {
          name: "Toyota Corolla 1.5v",
          cost: 1150.98
        },
        Display: {
          name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
          cost: 1500
        }
      }} />
      
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a different list when selected prop changes',() => {
    const wrapper = Enzyme.shallow(
      
      <FeaturePriceSummary 
      selected={{
        Processor: {
          name: "CHANGED PROP HERE",
          cost: 7000000
        },
        "Operating System": {
          name: "Ubuntu Linux 16.04",
          cost: 200
        },
        "Video Card": {
          name: "Toyota Corolla 1.5v",
          cost: 1150.98
        },
        Display: {
          name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
          cost: 1500
        }
      }} />
      
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });


});
