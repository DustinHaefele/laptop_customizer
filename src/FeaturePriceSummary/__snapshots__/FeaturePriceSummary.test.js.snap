// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Summary Component renders UI as expected 1`] = `
<section
  className="main__summary"
>
  <h3>
    NEW GREENLEAF 2018
  </h3>
  <div
    key="Processor"
  >
    <FeaturePrice
      cost={700}
      myKey="Processor"
      name="17th Generation Intel Core HB (7 Core with donut spare)"
    />
  </div>
  <div
    key="Operating System"
  >
    <FeaturePrice
      cost={200}
      myKey="Operating System"
      name="Ubuntu Linux 16.04"
    />
  </div>
  <div
    key="Video Card"
  >
    <FeaturePrice
      cost={1150.98}
      myKey="Video Card"
      name="Toyota Corolla 1.5v"
    />
  </div>
  <div
    key="Display"
  >
    <FeaturePrice
      cost={1500}
      myKey="Display"
      name="15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs"
    />
  </div>
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Your Price: 
    </div>
    <div
      className="summary__total__value"
    >
      $3,550.98
    </div>
  </div>
</section>
`;

exports[`Price Summary Component renders a different list when selected prop changes 1`] = `
<section
  className="main__summary"
>
  <h3>
    NEW GREENLEAF 2018
  </h3>
  <div
    key="Processor"
  >
    <FeaturePrice
      cost={7000000}
      myKey="Processor"
      name="CHANGED PROP HERE"
    />
  </div>
  <div
    key="Operating System"
  >
    <FeaturePrice
      cost={200}
      myKey="Operating System"
      name="Ubuntu Linux 16.04"
    />
  </div>
  <div
    key="Video Card"
  >
    <FeaturePrice
      cost={1150.98}
      myKey="Video Card"
      name="Toyota Corolla 1.5v"
    />
  </div>
  <div
    key="Display"
  >
    <FeaturePrice
      cost={1500}
      myKey="Display"
      name="15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs"
    />
  </div>
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Your Price: 
    </div>
    <div
      className="summary__total__value"
    >
      $7,002,850.98
    </div>
  </div>
</section>
`;
